platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java

```diff
@@ -53,5 +55,6 @@ private void prepareExecution() {
 
 		ISOCalendar.setDefaultTimeZone();
 		LogDb.setProcessClass(program.getClass());
+		CurrentUser.set(AGUser.getSystemUser());
```

Comment:
I think we should use `CurrentUserScope` in the surrounding `try` block instead. E.g.
```
try (var scope = new CurrentUserScope(AGUser.getSystemUser())) {
	prepareExecution();
	program.executeProgram();
}
```
That would prevent problems when `start()` gets called directly from a UI thread.

Reply:
I think the original solution was fine, considering the already existing code.

Reply:
I disagree. The proposed change eliminates a side effect which couly very easily be triggered by future code, while not really increasing complexity.